subset/alliedSwitch/src/main/java/alliedswitch/AlliedInterrogator.java

```diff
@@ -0,0 +1,462 @@
+        () -> {
+          parseData(data);
+        };
+    Thread parseThread = new Thread(runnable);
+
+    parseThread.start();
+  }
+
+  private void parse_packet(String raw_data, String[] show_expected, int[] show_pointers) {
+    int start = 0;
+    for (int port = 0; port < number_switch_ports; port++) {
+      for (int x = 0; x < show_expected.length; x++) {
+        start = recursive_data(raw_data, show_expected[x], start);
+        show_pointers[x] = start;
+      }
+
+      int chunk_s = show_pointers[0];
+      int chunk_e = show_pointers[3];
+
+      if (chunk_e == -1) chunk_e = raw_data.length();
+
+      String temp_data = raw_data.substring(chunk_s, chunk_e);
+
+      if (debug) System.out.println("length" + temp_data.length() + "temp_data:" + temp_data);
+
+      if (requestFlag == (interfacePos + 1)) {
+        parse_single(
+            temp_data, show_interface_expected, show_interface_pointers, show_interface_data);
+      } else if (requestFlag == (platformPos + 1)) {
+        parse_single(temp_data, show_platform_expected, show_platform_pointers, show_platform_data);
+      }
+    }
+  }
+
+  private void parse_single(
+      String raw_data, String[] expected_array, int[] pointers_array, String[] data_array) {
+    int start = 0;
+    for (int x = 0; x < expected_array.length; x++) {
+      start = recursive_data(raw_data, expected_array[x], start);
+      if (start == -1) {
+        start = pointers_array[x - 1];
+        pointers_array[x] = -1;
+      } else {
+        pointers_array[x] = start;
+      }
+    }
+
+    for (int x = 0; x < data_array.length; x++) {
+      int chunk_start = x * 2;
+      int chunk_end = 1 + (x * 2);
+
+      int chunk_s = pointers_array[chunk_start];
+      int chunk_e = pointers_array[chunk_end];
+
+      if (chunk_s > 0) {
+        if (chunk_e == -1) chunk_e = raw_data.length();
+
+        data_array[x] = raw_data.substring(chunk_s, chunk_e);
+
+        data_array[x] = data_array[x].substring(expected_array[chunk_start].length());
+
+        String extracted_data = expected_array[chunk_start] + data_array[x];
+
+        if (debug) System.out.println(extracted_data);
+
+        login_report += extracted_data + "\n";
+      }
+    }
+  }
+
+  private int recursive_data(String data, String search, int start) {
+    int pointer = data.indexOf(search, start);
+    if (debug) {
+      System.out.println("pointer:" + pointer);
+    }
+    return pointer;
+  }
+
+  public String getHostname() {
+    return device_hostname;
+  }
+
+  boolean userAuthorised = false;
+  boolean userEnabled = false;
+
+  public boolean getUserAuthorised() {
+    return userAuthorised;
+  }
+
+  public void setUserAuthorised(boolean userAuthorised) {
+    this.userAuthorised = userAuthorised;
+  }
+
+  public boolean getUserEnabled() {
+    return userEnabled;
+  }
+
+  public void setUserEnabled(boolean userEnabled) {
+    this.userEnabled = userEnabled;
+  }
+
+  private void parseData(String data) {
```

Comment:
this function is way too long to be understood. break it up into some sub functions...